### PR TITLE
VZ-5937.  Add known-issues checks/warnings hooks to BOM validator

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Add known-issues checks/warnings hooks to BOM validator
- Allows workaround for known issues to be tracked as warnings in the output
- Use for Rancher upgrade scenario where rancher-webhook image does not always update to what's in the BOM
- Add alternate tags list for rancher-webhook, warning message

Example output:

```
[c:kind-kind:bom-validator]$ bom-validator 
USING KUBECONFIG:  /home/mcico/.kube/config
The platform operator pod name is pod/verrazzano-platform-operator-57cf88464d-knscg

Images From BOM not installed in cluster
----------------------------------------
Image not installed: cert-manager-acmesolver:1.2.0-20220225033913-4f1415cb5
Image not installed: external-dns:v0.10.2-20220225102356-7bae1b96
Image not installed: rancher-agent:v2.5.9-20220506124723-e049e83ce
Image not installed: kubectl:v1.20.2
Image not installed: weblogic-monitoring-exporter:2.0.4

Image Warnings - Tags not at expected BOM level due to known issues
 ----------------------------------------
Warning: Image Name = fleet: Known issue for image fleet, found tag v0.3.5, expected tag v0.3.9 message: See VZ-5937
Warning: Image Name = gitjob: Known issue for image gitjob, found tag v0.1.15, expected tag v0.1.26 message: See VZ-5937
Warning: Image Name = rancher-webhook: Known issue for image rancher-webhook, found tag v0.1.2, expected tag v0.1.4 message: VZ-5937
Warning: Image Name = fleet-agent: Known issue for image fleet-agent, found tag v0.3.5, expected tag v0.3.9 message: See VZ-5937
```

Related to VZ-5937.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tools/bom-validator/bom-validator.go [hunk 1]
```diff
@@ -42,11 +51,27 @@ type imageError struct {
 
 var ignoreSubComponents = []string{}
 
+// Hack to work around an issue with the 1.2 upgrade; Rancher does not always update the webhook image
```
Comment (desagar) [inline tools/bom-validator/bom-validator.go:54]: maybe we should mention both the from and to versions we know the issue exists in - i.e. "the 1.2 -> 1.3 upgrade", and the JIRA number, which is context we may forget 6 months from now.

Comment (michael-cico) [reply tools/bom-validator/bom-validator.go:54]: you mean just hard-coded in the message text?  Otherwise I don't think the validator has the context.

Comment (michael-cico) [reply tools/bom-validator/bom-validator.go:54]: I think it affects 1.1 --> 1.2 also.

Comment (michael-cico) [reply tools/bom-validator/bom-validator.go:58]: updated and pushed

**Review by mgianatagh — DISMISSED** [review]

File: tools/bom-validator/bom-validator.go [hunk 2]
```diff
@@ -204,20 +249,31 @@ func ignoreSubComponent(name string) bool {
...
+		for name, msg := range warnings {
+			fmt.Printf("Warning: Image Name = %s: %s\n", name, msg)
+		}
 	}
+	fmt.Println()
 	// Dump Images that don't match BOM, Failure
-	if !isBOMValid {
-		fmt.Println("BOM Images that don't match Cluster Images")
-		fmt.Println("----------------------------------------")
+	if !errorsFound {
```
Comment (desagar) [inline tools/bom-validator/bom-validator.go:272]: Shouldn't this be `if errorsFound` ?

File: tools/bom-validator/bom-validator.go [hunk 3]
```diff
@@ -64,13 +93,13 @@ func main() {
...
-	isBOMValid := validateBOM(&vBom, imagesInstalled, imagesNotFound, imageTagErrors)
+	errorsFound := validateBOM(&vBom, imagesInstalled, imagesNotFound, imageTagErrors, imageWarnings)
 
 	// Write to stdout
-	reportResults(imagesNotFound, imageTagErrors, isBOMValid)
+	reportResults(imagesNotFound, imageTagErrors, imageWarnings, errorsFound)
 
 	// Failure
-	if !isBOMValid {
+	if !errorsFound {
```
Comment (desagar) [inline tools/bom-validator/bom-validator.go:102]: same q. as below - shouldn't this be `if errorsFound` ?

Comment (desagar) [reply tools/bom-validator/bom-validator.go:102]: Ok just re-read it - this name is misleading. validateBOM returns !errorsFound - both here and in reportResults, this variable should be renamed for readability.

File: tools/bom-validator/bom-validator.go (hunk 2 above)
Comment (desagar) [reply tools/bom-validator/bom-validator.go:272]: Just re-read this - see my earlier comment for details

Comment (michael-cico) [reply tools/bom-validator/bom-validator.go]: Fixed and pushed

**Review by desagar — APPROVED** [review]

**Pull request merged**